wp-content/themes/goonj-crm/functions.php

```diff
@@ -474,6 +494,27 @@ function goonj_redirect_after_individual_creation() {
 				);
 				break;
 			}
+			case 'dropping-center-contribution':
+				if ( ! $source ) {
+					\Civi::log()->warning('Source is missing for material contribution flow', ['individualId' => $_GET['individualId']]);
+					return;
+				}
+				// If the individual was created during a material contribution process,
+				// We need to determine from where they were attempting to contribute.
+	
+				// First, we check if the source of Individual is Dropping Center.
+				$droppingCenter = \Civi\Api4\EckEntity::get( 'Collection_Camp', false )
+					->addWhere( 'title', '=', $source )
+					->execute()->first();
```

Comment:
How do we make sure that this API call returns only a dropping center and not a collection camp?

Reply:
Based on the source, we retrieve the source from the URL of the  contribution of  landing page for dropping center  but, for safety, we can implement it like this:

````
$droppingCenter = \Civi\Api4\EckEntity::get( 'Collection_Camp', false )
					->addWhere( 'title', '=', $source )
					->addWhere('subtype:name', '=', 'Dropping_Center')
					->execute()->first();
```